sources/common/infoschema.go

```diff
@@ -74,7 +74,13 @@ func ProcessSchema(conv *internal.Conv, infoSchema InfoSchema) error {
 // If we can't get/process data for a table, we skip that table and process
 // the remaining tables.
 func ProcessData(conv *internal.Conv, infoSchema InfoSchema) {
-	for srcTable, srcSchema := range conv.SrcSchema {
+	// Tables are ordered in alphabetical order with one exception: interleaved
+	// tables appear after the population of their parent table.
+	orderTableNames := ddl.OrderTable(conv.SpSchema)
+
+	for _, spannerTable := range orderTableNames {
```

Comment:
In a very rare scenario, it could be possible that the next table in the list is a child of the current, and the first row of the next table references the last row of the current table. It could be possible that these are batched in a single mutation, and could potentially throw an error due to missing parent row. It is very rare to actually witness this, but I think it makes sense to make this code threadsafe and ensure we drain out the row queue and finish the writes before moving onto the next table in the for loop.

Reply:
You should be able to use bw.Flush() at the end to drain out the rows. If not, we might have to bubble the wait group counts uptil here.

Reply:
Making a clean cut and flushing on table change is a good way to do this. There is very little gain we will get optimizing on this, and the possible issues if we don't are numerous.

Reply:
done